asynch/connection.py

```diff
@@ -15,38 +15,46 @@ def __init__(
         user: str = constants.DEFAULT_USER,
         password: str = constants.DEFAULT_PASSWORD,
         host: str = constants.DEFAULT_HOST,
-        port: int = constants.DEFAULT_PORT,
+        port: int | None = None,
         database: str = constants.DEFAULT_DATABASE,
         cursor_cls=Cursor,
         echo: bool = False,
         stack_track: bool = False,
+        secure: bool = False,
         **kwargs,
     ):
+        if secure and not port:
+            port = constants.DEFAULT_SECURE_PORT
+        elif not secure and not port:
+            port = constants.DEFAULT_PORT
```

Comment:
```suggestion
        if port is None:
            port = constants.DEFAULT_SECURE_PORT if secure else constants.DEFAULT_PORT
```

If port is 0, maybe it is intentional, so preprocessing only None case for port.

Reply:
If you are OK with this, I don't see this part changed except `is None` checks. Am I wrong?